src/condor_gridmanager/arcresource.cpp

```diff
@@ -142,16 +154,18 @@ void ArcResource::DoPing( unsigned& ping_delay, bool& ping_complete,
 {
 	int rc;
 
-	if ( gahp->isInitialized() == false ) {
+	if ( gahp->isStarted() == false ) {
```

Comment:
This change seems unrelated to the rest?

Reply:
isInitialized() returns false if a proxy hasn't be set for the gahp_server. With token auth, there is no proxy, so we just want to wait until the gahp is running. We set the proxy immediately after gahp startup (in ArcJob), so it's okay to use isStarted() in all cases.